src/Esprima/Utils/KnRJavascriptTextWriter.cs

```diff
@@ -0,0 +1,480 @@
+﻿using System.Runtime.CompilerServices;
+using Esprima.Ast;
+
+namespace Esprima.Utils;
+
+/// <summary>
+/// Javascript text writer (code formatter) which implements the most common <see href="https://en.wikipedia.org/wiki/Indentation_style#K&amp;R_style">K&amp;R style</see>.
+/// </summary>
+public class KnRJavascriptTextWriter : JavascriptTextWriter
+{
+    public new record class Options : JavascriptTextWriter.Options
```

Comment:
the `new` here feels bad, should this just be `KnRJavascriptTextWriterOptions`? In deal world could just pass suitable `Options` instance to `JavascriptTextWriter` that configures KnR style, but seems that it wouldn't be that easy to implement.